Watchman.Discord/Areas/Users/Controllers/UsersController.cs

```diff
@@ -44,10 +49,24 @@ public void GetAvatar(DiscordRequest request, Contexts contexts)
         [DiscordCommand("add role")] //todo
         public void AddRole(DiscordRequest request, Contexts contexts)
         {
-            var commandRole = request.OriginalMessage.ToLowerInvariant().Replace("-add role ", string.Empty); //TODO use DiscordRequest properties
-            var safeRoles = this._queryBus.Execute(new GetDiscordServerSafeRolesQuery(contexts.Server.Id)).SafeRoles;
-            var messagesService = _messagesServiceFactory.Create(contexts);
-            _rolesService.AddRoleToUser(safeRoles, messagesService, contexts, commandRole);
+            var args = request.Arguments.Skip(1).ToList();
+            var rolesToAssign = args.Select(x => x.Value);
+
+            if (rolesToAssign.Count() < 1)
+            {
+                throw new NotEnoughArgumentsException();
+            }
+
+            if (args.HasDuplicates())
+            {
+                throw new ArgumentsDuplicatedException();
+            }
+
+            var safeRoleNames = _queryBus.Execute(new GetDiscordServerSafeRolesQuery(contexts.Server.Id)).SafeRoles
+                .ToList().Select(x => x.Name);
```

Comment:
ToList().Select() jest kompletnie bez sensu

Reply:
jak chcesz wrzucać do nowej linii, to zacznij tą linie od .SafeRoles a nie .ToList()